config/clusters/k8s-prow/cluster-configuration.tf

```diff
@@ -0,0 +1,172 @@
+  # Set to `false` to utilize RBAC.
+  enable_legacy_abac = true
+
+  master_auth {
+    username = ""
+    password = ""
+
+    client_certificate_config {
+      issue_client_certificate = false
+    }
+  }
+}
+
+
+# The "default-pool" pool is for running Prow service component nodes (e.g. Deck, Plank, Sinker, etc.)
+resource "google_container_node_pool" "default_nodes" {
+  name       = "default-pool"
+  location   = "${google_container_cluster.cluster.location}"
+  cluster    = "${google_container_cluster.cluster.name}"
+  node_count = 8
+
+  # Auto repair, and auto upgrade nodes to match the master version
+  management {
+    auto_repair  = false
+    auto_upgrade = true
+  }
+
+  node_config {
+    machine_type = "n1-standard-4"
+    disk_size_gb = "100"
+    labels = {
+      role = "prow"
+    }
+    oauth_scopes = [
+      # Compute Engine (rw)
+      "https://www.googleapis.com/auth/compute",
+      # Storage (full)
+      "https://www.googleapis.com/auth/devstorage.full_control",
+      # Storage (ro)
+      "https://www.googleapis.com/auth/devstorage.read_only",
+      # Service Control (enabled)
+      "https://www.googleapis.com/auth/servicecontrol",
+      # Service Management (rw)
+      "https://www.googleapis.com/auth/service.management",
+      # Service Management (ro)
+      "https://www.googleapis.com/auth/service.management.readonly",
+      # Stackdriver Logging (wo)
+      "https://www.googleapis.com/auth/logging.write",
+      # Stackdriver Monitoring (full)
+      "https://www.googleapis.com/auth/monitoring",
+    ]
+  }
+}
+
+# The "ghproxy" pool is for running the GitHub reverse proxy cache (i.e. GHproxy)
+resource "google_container_node_pool" "ghproxy_nodes" {
+  provider = "google-beta"
+
+  name       = "ghproxy"
+  location   = "${google_container_cluster.cluster.location}"
+  cluster    = "${google_container_cluster.cluster.name}"
+  node_count = 1
+
+
+  # Auto repair, and auto upgrade nodes to match the master version
+  management {
+    auto_repair  = true
+    auto_upgrade = true
+  }
+
+  #  The node configuration of the pool.
+  node_config {
+    machine_type = "n1-standard-8"
+    disk_size_gb = "100"
+    labels = {
+      dedicated = "ghproxy"
+    }
+    taint {
+      key    = "dedicated"
+      value  = "ghproxy"
+      effect = "NO_SCHEDULE"
+    }
+    oauth_scopes = [
+      # Compute Engine (rw)
+      "https://www.googleapis.com/auth/compute",
+      # Storage (ro)
+      "https://www.googleapis.com/auth/devstorage.read_only",
+      # Service Control (enabled)
+      "https://www.googleapis.com/auth/servicecontrol",
+      # Service Management (rw)
+      "https://www.googleapis.com/auth/service.management",
+      # Stackdriver Logging (wo)
+      "https://www.googleapis.com/auth/logging.write",
+      # Stackdriver Monitoring (full)
+      "https://www.googleapis.com/auth/monitoring",
+    ]
+  }
+}
+
+resource "google_container_node_pool" "n1_standard_8_nodes" {
```

Comment:
is this the "trusted builds" pool now?

Reply:
My knowledge on the cluster structure is limited. It does have the name `n1-standard-8` and appears to have both prow components **and** jobs running in it.

Reply:
I do not see any labels specifying otherwise, so the "trusted builds" would be on this node pool or the `default-pool`.

Reply:
If we want to change this behavior we can taint the `default-pool` and use node selectors we can but this is current behavior from my understanding.

Reply:
the prow components could also use node-selectors to pin to their cluster.

cc @cjwagner

Reply:
I think it would be a good idea to keep trusted builds on a pool other than the Prow services (or possibly another cluster for that matter...) if only to avoid some heavy build suddenly starving out one of the services, but ghproxy probably doesn't need a dedicated node pool.

I vaguely remember the dedicated node pool pattern being borrowed from greenhouse, but we're serving a LOT more traffic there and it's probably unnecessary for ghrpxoy.

Greenhouse hit > 500 MB/s outgoing traffic, > 200 MB/s incoming traffic, and well over 1000 disk IOPS in the past 12 hours which somewhat justifies dedicating a node, since unfortunately I/O is not really schedulable and we'd pass the GCE limits on the smaller nodes IIRC.

ghproxy seems to be entirely < 2 MB/s in and < 2 MB/s out and under 40 IOPS. 

(I just spot checked these).

Reply:
> possibly another cluster
👍 

> heavy build suddenly starving out one of the services

Resource limits will prevent this scenario right? It seems like kubernetes is doing something wrong if we need separate clusters per task because we can't isolate them.

PS: also Travis I'm super happy to check in something that reflects the current state of the world. Being able to have these conversation -- and refer to them a year later -- is the whole reason this file is awesome. So this is mostly getting me to think about how we should design the cluster, rather than something to change in this file.